### ISSUE TITLE
keydown error due to conflict with nya-bootstrap-select plugin

### ISSUE THREAD

**Issue by szmalec** [snapappointments/bootstrap-select#1710]
I'am using _bootstrap-select_ AND _[nya-bootstrap-select](https://github.com/lordfriend/nya-bootstrap-select)_ on the same page. When I press key down when in searchbox of _nya-bootstrap-select_ then _keydown_ event handler from _bootstrap-select_ is raised which leads to error:

> Uncaught TypeError: Cannot read property 'options' of undefined

Error occurs in following line of _keydown_ handler:

`if (that.options.liveSearch) $parent = $this.parent().parent();`

it occurs because this

`that = $parent.data('this')`

- _that_ is assigned to _undefined_ 

Problem probably occurs because control is NOT _bootstrap-select_ - it is _nya-bootstrap-select_ which uses _bs-searchbox_ css style for searchbox and as we can see in _bootstrap-select_ source code, this style is used in selector for keydown handler:

```
  $(document)
      .data('keycount', 0)
      .on('keydown', '.bootstrap-select [data-toggle=dropdown], .bootstrap-select [role="menu"], .bs-searchbox input', Selectpicker.prototype.keydown)
      .on('focusin.modal', '.bootstrap-select [data-toggle=dropdown], .bootstrap-select [role="menu"], .bs-searchbox input', function (e) {
        e.stopPropagation();
      });

```

Can you provide fix for this? I think that in code above, selector should be modified to something like this:

```
  $(document)
      .data('keycount', 0)
      .on('keydown', '.bootstrap-select [data-toggle=dropdown], .bootstrap-select [role="menu"], .bootstrap-select .bs-searchbox input', Selectpicker.prototype.keydown)
      .on('focusin.modal', '.bootstrap-select [data-toggle=dropdown], .bootstrap-select [role="menu"], .bootstrap-select .bs-searchbox input', function (e) {
        e.stopPropagation();
      });

```

**Comment by caseyjhol** [comment]
Possibly fixed now that the keydown event is namespaced?

**Comment by iantropov** [comment]
Hi @caseyjhol 

> Possibly fixed now that the keydown event is namespaced?

No, because namespacing of event doesn't affect subscription to `keydown` event. And now we subscribe to *all* `input` in `.bs-searchbox` containers (https://github.com/silviomoreto/bootstrap-select/blob/master/js/bootstrap-select.js#L1833), but some of them can be placed outside bootstrap select component (may be inside other plugins, like mentioned nya-bootstrap-select).

I think, that we should re-open this issue. Also, if you want, I can prepare PR for this.

**Comment by caseyjhol** [comment]
Released in [v1.13.4](https://github.com/snapappointments/bootstrap-select/releases/tag/v1.13.4)!
